### PR TITLE
Feature search feed

### PR DESCRIPTION
**Pull request by OmKakatkar**
# Feature : User Profile, Search User, Sorted Feed

## Additions
- User can be search from search bar.
- Individual user profile can be accessed
- Feed can be sorted by Trending, New and Personalized

### CODE REVIEW

File: src/frontend/components/Header/Header.jsx [hunk 1]
```diff
@@ -1,15 +1,50 @@
...
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useEffect } from "react";
+import { useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
-import { NavLink, useNavigate } from "react-router-dom";
+import { Link, NavLink, useLocation, useNavigate } from "react-router-dom";
 
 import { handleLogout } from "../../features/Auth/authSlice";
+import useDetectClickOutside from "../../hooks/useDetectClickOutside";
+import { debounce } from "../../utils/util";
```
Comment (srv-pranjal) [inline src/frontend/components/Header/Header.jsx:10]: Suggestion: try using absolute imports to avoid "../../"

**Review by Amandeep-sinha — APPROVED** [review]
UI looks awesome....
Good to go ahead

File: src/frontend/components/Header/Header.jsx [hunk 2]
```diff
@@ -1,15 +1,50 @@
...
+	const { allUsers } = useSelector((store) => store.user);
+	const [searchUser, setSearchUser] = useState("");
+	const { triggerRef, nodeRef, showItem, setShowItem } =
+		useDetectClickOutside();
+	const [foundUsers, setFoundUsers] = useState([]);
+	const location = useLocation();
+
+	useEffect(() => {
+		if (searchUser !== "") {
+			debounce(() => {
```
Comment (srv-pranjal) [inline src/frontend/components/Header/Header.jsx:27]: great to see that you have used debounce to optimize search

File: src/frontend/components/Header/Header.jsx (hunk 1 above)
Comment (khushijohri2001) [reply src/frontend/components/Header/Header.jsx:10]: I would suggest the same. It was giving an error while i was deploying it to Netlify then I had to change the imports

File: src/frontend/features/Profile/Profile.jsx [hunk 3]
```diff
@@ -1,19 +1,28 @@
 import { useSelector } from "react-redux";
+import { useParams } from "react-router-dom";
 import ModalCard from "../../components/ModalCard/ModalCard";
 import ProfileUpdate from "../../components/ProfileUpdate/ProfileUpdate";
```
Comment (srv-pranjal) [inline src/frontend/features/Profile/Profile.jsx:5]: Suggestion: Barrel Exports can be a good idea here to streamline and simplify imports-exports

**Review by srv-pranjal — APPROVED** [review]
Left some suggestions.
Rest everything looks modular.
Good work

File: src/frontend/components/Header/Header.jsx (hunk 1 above)
Comment (OmKakatkar) [reply src/frontend/components/Header/Header.jsx:10]: Thanks for the suggestions. I will look into absolute imports

File: src/frontend/features/Profile/Profile.jsx (hunk 3 above)
Comment (OmKakatkar) [reply src/frontend/features/Profile/Profile.jsx:5]: Thanks for the suggestion, I will look into Barrel Exports as well

**Pull request merged**